From a96726a1ea141874b64a6fd1bd927380c3c833c0 Mon Sep 17 00:00:00 2001
From: Piotr <piotr1212@gmail.com>
Date: Mon, 12 Nov 2018 11:10:42 +0100
Subject: [PATCH] Add locking in cache.py

fixes: #822 #815
---
 lib/carbon/cache.py | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/lib/carbon/cache.py b/lib/carbon/cache.py
index 23bb11d..da873d1 100644
--- a/lib/carbon/cache.py
+++ b/lib/carbon/cache.py
@@ -158,13 +158,16 @@ class _MetricCache(defaultdict):
 
   @property
   def counts(self):
-    return [(metric, len(datapoints)) for (metric, datapoints) in self.items()]
+    with self.lock:
+      return [(metric, len(datapoints)) for (metric, datapoints)
+              in self.items()]
 
   @property
   def watermarks(self):
-    return [(metric, min(datapoints.keys()), max(datapoints.keys()))
-            for (metric, datapoints) in self.items()
-            if datapoints]
+    with self.lock:
+      return [(metric, min(datapoints.keys()), max(datapoints.keys()))
+              for (metric, datapoints) in self.items()
+              if datapoints]
 
   @property
   def is_full(self):
-- 
2.20.1

