Description: change default values for debian
Author: Jonas Genannt <jonas.genannt@capi2name.de>
Forwarded: not-needed

--- a/conf/carbon.conf.example
+++ b/conf/carbon.conf.example
@@ -30,14 +30,20 @@
 #
 #LOCAL_DATA_DIR = /opt/graphite/storage/whisper/
 
+STORAGE_DIR    = /var/lib/graphite/
+LOCAL_DATA_DIR = /var/lib/graphite/whisper/
+CONF_DIR       = /etc/carbon/
+LOG_DIR        = /var/log/carbon/
+PID_DIR        = /var/run/
+
 # Enable daily log rotation. If disabled, carbon will automatically re-open
 # the file if it's rotated out of place (e.g. by logrotate daemon)
-ENABLE_LOGROTATION = True
+ENABLE_LOGROTATION = False
 
 # Specify the user to drop privileges to
 # If this is blank carbon runs as the user that invokes it
 # This user must have write access to the local data directory
-USER =
+USER = _graphite
 #
 # NOTE: The above settings must be set under [relay] and [aggregator]
 #       to take effect for those daemons as well
